src/services/completions.ts

```diff
@@ -3051,4 +3054,9 @@ namespace ts.Completions {
             addToSeen(seenModules, getSymbolId(sym)) &&
             checker.getExportsOfModule(sym).some(e => symbolCanBeReferencedAtTypeLocation(e, checker, seenModules));
     }
+
+    function isDeprecated(symbol: Symbol, checker: TypeChecker) {
+        const declarations  = skipAlias(symbol, checker).declarations;
+        return length(declarations) && every(declarations, d => !!(getCombinedNodeFlagsAlwaysIncludeJSDoc(d) & ModifierFlags.Deprecated));

```

Comment:
Can this be?
```suggestion
        return skipAlias(symbol, checker).declarations?.every(isDeprecatedDeclaration );
```

Reply:
For an empty array, `every` returns `true` - for that reason need to check the `length` of an array. Does that make sense?